### DISCUSSION TITLE
Login button

### DISCUSSION THREAD

**Discussion by rmuhawieh** [imartinez/privateGPT#1229] [General]
Say I wanted to implement a login button where a user login would open a page with less features than say an admin login, how would I go about that. I've tried utilizing the gr.Button() but the Button.click() doesn't seem to work for me

**Comment by 888kr** [comment]
its not simple. But you can keep the same window, but instead for the various functions, you want to check if a user has a state or constant, i.e "isAdmin=False", and then allow certain features. I'd recommend doing this across a server where script can't be tempered client-sided by simply changing a boolean. (don't take this as professional advice)
